building_sim_plugins/building_gazebo_plugins/src/crowd_simulator.cpp

```diff
@@ -125,26 +134,28 @@ void CrowdSimulatorPlugin::_update_internal_object(
   if (!object_ptr)
   {
     RCLCPP_ERROR(
-      _crowd_sim_interface->logger(), "Null objectPtr when update Object!");
+      crowd_sim_interface->logger(), "Null objectPtr when update Object!");
     return;
   }
   if (!model_ptr)
   {
     RCLCPP_ERROR(
-      _crowd_sim_interface->logger(), "Null modelPtr when update Object!");
+      crowd_sim_interface->logger(), "Null modelPtr when update Object!");
     return;
   }
 
   //update pose from menge to gazebo
   ignition::math::Pose3d pose =
-    _crowd_sim_interface->get_agent_pose<ignition::math::Pose3d>(
+    crowd_sim_interface->get_agent_pose<ignition::math::Pose3d>(
     object_ptr->agent_ptr,
     delta_sim_time);
 
   gazebo::physics::ActorPtr actor_ptr =
     boost::dynamic_pointer_cast<gazebo::physics::Actor>(model_ptr);
 
-  auto delta_dist_vector = pose.Pos() - actor_ptr->WorldPose().Pos();
+  auto pos_old = actor_ptr->WorldPose().Pos();
+  auto pos_new = pose.Pos();
+  auto delta_dist_vector = pos_new - pos_old;
```

Comment:
I was fiddling with some pose values and forgot to undo the change.

Reply:
pos_old is used below for shorter ref name.

Reply:
Yea but I'm not even sure what the pos_old below is used for? I find the logic a bit hard to follow, `anim_pose` seems to be set with a mix of `init_pose`, `pose`, and `old_pose`? Can't one of those be used instead of `old_pose`?

Reply:
Ok, I have reverted to the original version mostly. The minor difference being singling out original z position in world pose, save it, and set it again after many other operations before the final `SetWorldPose` statement.